lib/openstack/amqp/openstack_qpid_connection.rb

```diff
@@ -71,9 +71,24 @@ def connection # :nodoc:
   end
 
   def create_connection
-    url = "#{@options[:hostname]}:#{@options[:port]}"
-    opts = @options.select { |k, v| [:username, :password].include? k }
-    opts[:transport] = "ssl" if @options[:port] == 5671
-    Qpid::Messaging::Connection.new(:url => url, :options => opts)
+    Qpid::Messaging::Connection.new(:url => connection_url, :options => connection_options)
+  end
+
+  def connection_options
+    connection_opts = {}
+    if @options[:port] == "5671"
+      connection_opts[:transport]     = "ssl"
+      connection_opts[:ssl_cert_name] = @options[:hostname]
+    end
+    if @options.key?(:username)
+      connection_opts[:sasl_mechanism] = "PLAIN"
+      connection_opts[:username]       = @options[:username]
+      connection_opts[:password]       = @options[:password]
+    end
+    connection_opts
+  end
+
+  def connection_url
+    "#{@options[:hostname]}:#{@options[:port]}"
```

Comment:
hrm, under the covers, the qpid library is *probably* using:

```ruby
  amqp://user/password@host:port
```

I know I've tried that format directly in the URL, but it fails with the `qpid_messaging` gem.  I could see if there's a `:port` option that I could pass, but I doubt it.  And, even if there is, I doubt it works. (so pessimism much cynical)